docs/design.html

```diff
@@ -313,8 +313,8 @@ <h3><a id="replication" href="#replication">4.7 Replication</a></h3>
     have the option of either waiting for the message to be committed or not, depending on their preference for tradeoff between latency and durability. This preference is controlled by the acks setting that the
     producer uses.
     Note that topics have a setting for the "minimum number" of in-sync replicas that is checked when the producer requests acknowledgment that a message
-    has been written to the full set of in-sync replicas. If a less stringent acknowledgement is requested by the producer, then the message can be committed, and consumed,
-    even if the number of in-sync replicas is lower than the minimum (e.g. it can be as low as just the leader).
+    has been written to the full set of in-sync replicas. If a less stringent acknowledgement is requested by the producer, the message will be acknowledged early
+    but will not be committed, nor consumed, until all in-sync replicas apply it to their log.
     <p>
```

Comment:
suggested revision:

```
A message is considered committed when all in-sync replicas for that partition have applied the message 
to their logs. Because only committed messages are given out to the consumer. The consumer will not see 
a message that could be lost if the leader fails. Producers have the option of either waiting for the message 
to be committed or not, depending on their latency and durability settings. This behavior is controlled by the 
producer acks setting. Topics have a setting for the "minimum number" of in-sync replicas. This setting is 
checked when the producer requests acknowledgment that a message has been written to the full set of in-sync 
replicas. If a less stringent acknowledgement is requested by the producer, the message is acknowledged early 
but is not committed or consumed until the in-sync replicas apply it to their log.
```

Reply:
This seems good to me, one suggestion:

"Because only committed messages are given out to the consumer. The consumer will not see 
a message that could be lost if the leader fails" -> "Because only committed messages are given out to the consumer, the consumer will not see a message that could be lost if the leader fails."